components/camel-netty-http/src/main/java/org/apache/camel/component/netty/http/handlers/HttpServerChannelHandler.java

```diff
@@ -316,11 +318,44 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
 
     @Override
     protected Object getResponseBody(Exchange exchange) throws Exception {
+        HttpResponse response;
+        Message answer;
         // use the binding
         if (exchange.hasOut()) {
-            return consumer.getEndpoint().getNettyHttpBinding().toNettyResponse(exchange.getOut(), consumer.getConfiguration());
+            response = consumer.getEndpoint().getNettyHttpBinding().toNettyResponse(exchange.getOut(), consumer.getConfiguration());
+            answer = exchange.getOut();
         } else {
-            return consumer.getEndpoint().getNettyHttpBinding().toNettyResponse(exchange.getIn(), consumer.getConfiguration());
+            response = consumer.getEndpoint().getNettyHttpBinding().toNettyResponse(exchange.getIn(), consumer.getConfiguration());
+            answer = exchange.getIn();
         }
+        
+        // TODO is this where we should tackle handling the 204 change?
+        handleNoContent(exchange, answer, response);
```

Comment:
Do you think if better handle the status code here: https://github.com/apache/camel/blob/master/components/camel-netty-http/src/main/java/org/apache/camel/component/netty/http/DefaultNettyHttpBinding.java#L394, because that is the method sole responsibility to produce the correct HTTP response and thus you won't need any further logic elsewhere

Reply:
working on that change now...

Reply:
should be good to go